From 2860d4c6927b7bf4c6b494aa0ee43fb467253bce Mon Sep 17 00:00:00 2001
From: samaga Prasanna Krishna <samaga.krishna@intel.com>
Date: Wed, 27 Mar 2013 16:10:17 +0530
Subject: [PORT FROM MAIN]Audio:Restoring the file duration during suspend

BZ: 94542

During offload suspend, need to store the file duration. After
reset during offload revert back the value of file duration stored.
Screan off/on or rotation of the screen after long pause ( offload
suspend) the application will call getDuration function of
AwesomePlayer. This function has to return the store value instead
of -1. The code changes are done to address this issue.

Change-Id: Idc927eb736d3e1c62b980386e11bfc484ad3ce2d
Orig-Change-Id: Ib85f0099a52505e91b9e4859238a654bb015e1e5

Signed-off-by: Arulselvan M <arulselvan.m@intel.com>
---
 media/libstagefright/AwesomePlayer.cpp       |    3 ++-
 media/libstagefright/include/AwesomePlayer.h |    2 +-
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/media/libstagefright/AwesomePlayer.cpp b/media/libstagefright/AwesomePlayer.cpp
index 13e4847..46810da 100644
--- a/media/libstagefright/AwesomePlayer.cpp
+++ b/media/libstagefright/AwesomePlayer.cpp
@@ -3105,6 +3105,7 @@ status_t AwesomePlayer::offloadSuspend() {
     stats.mFlags = mFlags & (PLAYING | AUTO_LOOPING | LOOPING | AT_EOS);
     getPosition(&stats.mPositionUs);
     mOffloadPauseUs = stats.mPositionUs;
+    stats.mDurationUs = mDurationUs; /* store the file duration */
     extractorFlags = mExtractorFlags;
     if (mOffload && ((mFlags & PLAYING) == 0)) {
          ALOGV("offloadSuspend(): Deleting timer");
@@ -3116,7 +3117,7 @@ status_t AwesomePlayer::offloadSuspend() {
     }
 
     reset_l();
-
+    mDurationUs = stats.mDurationUs; /* restore the duration */
     mExtractorFlags = extractorFlags;
     mStats = stats;
     return OK;
diff --git a/media/libstagefright/include/AwesomePlayer.h b/media/libstagefright/include/AwesomePlayer.h
index bd31396..cb9b981 100644
--- a/media/libstagefright/include/AwesomePlayer.h
+++ b/media/libstagefright/include/AwesomePlayer.h
@@ -333,7 +333,7 @@ private:
         KeyedVector<String8, String8> mUriHeaders;
         sp<DataSource> mFileSource;
         int64_t mPositionUs;
-
+        int64_t mDurationUs; /* store the file duration */
         // FIXME:
         // These two indices are just 0 or 1 for now
         // They are not representing the actual track
-- 
1.7.4.1

